[Bug]: Secret files not working anymore

### What happened?

Starting with version v0.16.5.8, it seems like both the `SPARKY_FITNESS_API_ENCRYPTION_KEY_FILE` and `BETTER_AUTH_SECRET_FILE` env variables are not working anymore.
The only way to provide these 2 secrets is via environment variables, however I could not find any deprecation notice.

Unless `BETTER_AUTH_SECRET` is explicitly defined, I get:

```text
[dotenv@17.3.1] injecting env (0) from ../.env -- tip: 🔐 encrypt with Dotenvx: https://dotenvx.com
node:buffer:322
  throw new ERR_INVALID_ARG_TYPE(
        ^

TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received undefined
    at Function.from (node:buffer:322:9)
    at <anonymous> (/app/SparkyFitnessServer/auth.ts:157:18)
    at ModuleJob.run (node:internal/modules/esm/module_job:325:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:606:24)
    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:117:5) {
  code: 'ERR_INVALID_ARG_TYPE'
}

Node.js v20.20.2
```

in the Server logs.
To be fair, the docs don't show anymore the possibility to use `BETTER_AUTH_SECRET_FILE`, but I was running up until version v0.16.5.7 without any issues using the file.

Then, even if I correctly set `BETTER_AUTH_SECRET`, but still keep `SPARKY_FITNESS_API_ENCRYPTION_KEY_FILE`, I run into this other error:

```text
[dotenv@17.3.1] injecting env (0) from ../.env -- tip: ⚙️  override existing env vars with { override: true }
[2026-04-19T17:09:18.808Z] [ERROR] Encryption key (SPARKY_FITNESS_API_ENCRYPTION_KEY) is missing. Update your environment variable and try again.
/app/SparkyFitnessServer/security/encryption.ts:9
  throw new Error('SPARKY_FITNESS_API_ENCRYPTION_KEY is missing.');
        ^


Error: SPARKY_FITNESS_API_ENCRYPTION_KEY is missing.
    at <anonymous> (/app/SparkyFitnessServer/security/encryption.ts:9:9)
    at ModuleJob.run (node:internal/modules/esm/module_job:325:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:606:24)
    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:117:5)

Node.js v20.20.2
```

which is unexpected, as I am passing the encryption key via file.

Both errors are critical and prevent the Server from properly starting up.

### Steps to reproduce

1. Deploy Sparkyfitness v0.16.5.8 passing the API encryption key and BetterAuth secret using files (by setting the env variables `SPARKY_FITNESS_API_ENCRYPTION_KEY_FILE` and `BETTER_AUTH_SECRET_FILE`)
2. Observe logs
3. Verify that it is not possible to access Sparkyfitness

### Expected behavior

The secrets are extracted from the files.

### How are you running SparkyFitness?

Other

### SparkyFitness version

v0.16.5.8

### Which area is affected?

Server / API

### Relevant logs or screenshots

```shell

```

### Anything else?

Running on K8s (with custom manifests - no Helm chart).
Here is the Deployment definition:

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: sparkyfitness-backend
  namespace: sparkyfitness
  labels:
    app: sparkyfitness-backend
spec:
  replicas: 1
  selector:
    matchLabels:
      app: sparkyfitness-backend
  strategy: # Downtime at rollout, but clean PV release
    type: Recreate
  revisionHistoryLimit: 3
  template:
    metadata:
      labels:
        app: sparkyfitness-backend
    spec:
      automountServiceAccountToken: false
      dnsPolicy: ClusterFirst
      dnsConfig:
        searches:
          - sparkyfitness.svc.cluster.local
      containers:
        - name: sparkyfitness-backend
          image: codewithcj/sparkyfitness_server:v0.16.5.8
          env:
            - name: SPARKY_FITNESS_LOG_LEVEL
              value: WARN
            - name: SPARKY_FITNESS_DB_HOST
              value: sparkyfitness-db-rw
            - name: SPARKY_FITNESS_DB_PORT
              value: "5432"
            - name: SPARKY_FITNESS_DB_NAME
              value: sparkyfitness
            - name: SPARKY_FITNESS_DB_USER
              value: sparky
            - name: SPARKY_FITNESS_DB_PASSWORD
              valueFrom:
                secretKeyRef:
                  name: sparkyfitness-cnpg-secret
                  key: password
            - name: SPARKY_FITNESS_APP_DB_USER
              value: sparkyapp
            - name: SPARKY_FITNESS_APP_DB_PASSWORD
              valueFrom:
                secretKeyRef:
                  name: sparkyfitness-app-cnpg-secret
                  key: password
            - name: SPARKY_FITNESS_API_ENCRYPTION_KEY_FILE
              value: /run/secrets/sparkyfitness_api_key
            - name: BETTER_AUTH_SECRET_FILE
              value: /run/secrets/sparkyfitness_better_auth_secret
            - name: SPARKY_FITNESS_FRONTEND_URL
              value: <REDACTED>
            - name: ALLOW_PRIVATE_NETWORK_CORS
              value: "false"
            - name: SPARKY_FITNESS_EXTRA_TRUSTED_ORIGINS
              value: <REDACTED>
            - name: TZ
              value: Europe/Amsterdam
          resources:
            requests:
              cpu: 100m
              memory: 256Mi
            limits:
              cpu: 800m
              memory: 512Mi
          volumeMounts:
            - mountPath: /app/SparkyFitnessServer/backup
              subPath: backup
              name: sparkyfitness-backend
            - mountPath: /app/SparkyFitnessServer/uploads
              subPath: uploads
              name: sparkyfitness-backend
            - mountPath: /run/secrets
              name: secret-files
              readOnly: true
      volumes:
        - name: sparkyfitness-backend
          persistentVolumeClaim:
            claimName: sparkyfitness-backend
        - name: secret-files
          secret:
            secretName: sparkyfitness-secrets
```

--- a/SparkyFitnessServer/SparkyFitnessServer.ts
+++ b/SparkyFitnessServer/SparkyFitnessServer.ts
@@ -2,8 +2,6 @@ import './env.js'; // Diese Zeile MUSS ganz oben stehen
 import path from 'path';
 
 import fs from 'fs';
-import { loadSecrets } from './utils/secretLoader.js';
-import { runPreflightChecks } from './utils/preflightChecks.js';
 import express from 'express';
 // @ts-expect-error TS7016
 import cors from 'cors';
@@ -103,13 +101,6 @@ import { fileURLToPath } from 'url';
 const __filename = fileURLToPath(import.meta.url);
 const __dirname = path.dirname(__filename);
 
-loadSecrets();
-try {
-  runPreflightChecks();
-} catch (error) {
-  process.exitCode = 1;
-  throw error;
-}
 const app = express();
 app.set('trust proxy', 1); // Trust the first proxy immediately in front of me just internal nginx. external not required.
 const PORT = process.env.SPARKY_FITNESS_SERVER_PORT || 3010;

--- a/SparkyFitnessServer/env.ts
+++ b/SparkyFitnessServer/env.ts
@@ -1,6 +1,19 @@
 import path from 'path';
 import { fileURLToPath } from 'url';
 import dotenv from 'dotenv';
+import { loadSecrets } from './utils/secretLoader.js';
+import { runPreflightChecks } from './utils/preflightChecks.js';
 
 const __dirname = path.dirname(fileURLToPath(import.meta.url));
 dotenv.config({ path: path.resolve(__dirname, '../.env') });
+
+// Load secrets from files (Docker Swarm / Kubernetes support)
+loadSecrets();
+
+// Run pre-flight checks for essential environment variables
+try {
+  runPreflightChecks();
+} catch (error) {
+  process.exitCode = 1;
+  throw error;
+}

